brave/brave-core
# (Wallet) Implement NFT asset details screen

## Resolves https://github.com/brave/brave-browser/issues/27276

Implements new NFT asset detail screen supporting GIF, BMP, PNG and JPG images.

https://user-images.githubusercontent.com/3308503/212389665-46eaee7c-9957-47b2-8a95-dd118942caa4.mov

## Security Review

Requested security review: https://github.com/brave/security/issues/1168.

## Light Theme

<img width="377" alt="Android_SDK_built_for_x86_and_Comparing_master___simone_nft-detail-screen_·_brave_brave-core" src="https://user-images.githubusercontent.com/3308503/212390078-80d15532-3caf-4574-8b8a-ad0b19ca4518.png">

## Dark Theme

<img width="376" alt="Android_SDK_built_for_x86" src="https://user-images.githubusercontent.com/3308503/212390308-be14a13c-7a71-4b1f-8b1a-e2746ca528e6.png">


## Related Issues

- [Add support for Solana NFT in NFT detail screen](https://github.com/brave/brave-browser/issues/27803)
- [Show "Send" button in NFT detail screen for owned NFTs](https://github.com/brave/brave-browser/issues/27802)
- [Add SVG support to NFT images](https://github.com/brave/brave-browser/issues/27800)



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Open the wallet section
- If not already present, add a new NFT selecting "Edit visible assets" > "Add custom assets"
(E.g `0x59468516a8259058bad1ca5f8f4bff190d30e066`, id `183` -[ Invisible Friends](https://opensea.io/assets/ethereum/0x59468516a8259058bad1ca5f8f4bff190d30e066/183))
- Tap on the NFT item
- Observe the new NFT detail screen loads correctly image, title and description.
